nixos/modules/services/misc/rmfakecloud.nix

```diff
@@ -0,0 +1,119 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.rmfakecloud;
+  serviceDataDir = "/var/lib/rmfakecloud";
+
+in {
+  options = {
+    services.rmfakecloud = {
+      enable = mkEnableOption "rmfakecloud remarkable self-hosted cloud";
+
+      storageUrl = mkOption {
+        type = types.str;
+        example = "https://local.appspot.com";
+        description = ''
+          Controls whether file upload/download goes through the local proxy
+          or to an external server. It's the address of rmfakecloud as visible
+          from the tablet, especially if the host is behind a reverse proxy or
+          in a container.
+        '';
+      };
+
+      port = mkOption {
+        type = types.port;
+        default = 3000;
+        description = ''
+          Listening port number.
+        '';
+      };
+
+      logLevel = mkOption {
+        type = types.enum [ "info" "debug" "warn" "error" ];
+        default = "info";
+        description = ''
+          Set the log verbosity.
+        '';
+      };
+
+      extraSettings = mkOption {
+        type = with types; attrsOf str;
+        default = { };
+        example = { RM_HTTPS_COOKIE = true; };
+        description = ''
+          Extra settings in the form of a set of key-value pairs.
+          For tokens and secrets, use `environmentFile` instead.
+        '';
+      };
+
+      environmentFile = mkOption {
+        type = with types; nullOr path;
+        default = null;
+        example = "/etc/secrets/rmfakecloud.env";
+        description = ''
+          Path to an environment file.
+          This can be used to securely store tokens and secrets outside of the
+          world-readable Nix store.
+        '';
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+    systemd.services.rmfakecloud = {
+      description = "rmfakecloud remarkable self-hosted cloud";
+
+      environment = {
+        STORAGE_URL = cfg.storageUrl;
+        PORT = toString cfg.port;
+        LOGLEVEL = cfg.logLevel;
+      } // cfg.extraSettings;
+
+      preStart = ''
+        # Generate the secret key used to sign client session tokens.
+        # Replacing it invalidates the previously established sessions.
+        if [ ! -f jwt_secret_key ]; then
+          (umask 077; touch jwt_secret_key)
+          ${pkgs.openssl}/bin/openssl rand -base64 48 >> jwt_secret_key
+        fi
+      '';
+
+      script = ''
+        if [[ ! -v JWT_SECRET_KEY ]]; then
+          export JWT_SECRET_KEY=$(cat jwt_secret_key)
+        fi
+
+        ${pkgs.rmfakecloud}/bin/rmfakecloud
```

Comment:
I think the package used by the module should be an overridable option: The package proposed here does not include the user SPA (which is fine), but shall users of this module want to build it with the app, they should be able to write such derivation and have this module use it.

Reply:
The `rmfakecloud` package should eventually be completed to include the web UI as well. I do not think that there is any reason to propose different variants of the package, so I do not think that a `package` option in the module would be useful.

Reply:
I think some users may still prefer to have the package without the SPA (myself included), so imho, the package should eventually include an argument to choose whether the SPA should be compiled. It follows that an option package would probably be useful.

Reply:
Okay, I added a `package` option, since this does not cost anything.